classes/migration/upgrade/v3_6_0/I857_ContributorRolesTypes.php

```diff
@@ -0,0 +1,174 @@
+        Schema::create('contributor_role_settings', function (Blueprint $table) {
+            $table->comment('Contributor role settings');
+            $table->bigInteger('contributor_role_setting_id')->autoIncrement();
+            $table->bigInteger('contributor_role_id');
+            $table->string('setting_name', 255);
+            $table->string('setting_value', 255);
+            $table->string('locale', 28);
+            $table->unique(['contributor_role_id', 'setting_name', 'locale'], 'contributor_role_id_setting_name_locale_unique');
+            $table->foreign('contributor_role_id', 'contributor_role_id_settings_foreign')->references('contributor_role_id')->on('contributor_roles')->onDelete('cascade');
+        });
+
+        // Add foreign key
+        Schema::table('credit_contributor_roles', function (Blueprint $table) {
+            $table->foreign('contributor_role_id', 'contributor_role_id_foreign')->references('contributor_role_id')->on('contributor_roles')->onDelete('cascade');
+        });
+
+        // Add author user group if not exist
+        UserGroup::withRoleIds([Role::ROLE_ID_AUTHOR])
+            ->get()
+            ->groupBy(fn (UserGroup $usergroup) => $usergroup->context_id)
+            ->reject(fn ($group) => $group->contains(fn (UserGroup $ug) => $ug->nameLocaleKey === 'default.groups.name.author')) // Select where not author
+            ->each(function ($g, int $contextId) {
+                $supportedFormLocales = json_decode(DB::table('journal_settings')
+                    ->where('journal_id', '=', $contextId)
+                    ->where('setting_name', '=', 'supportedFormLocales')
+                    ->select('setting_value')
+                    ->get()
+                    ->pluck('setting_value')
+                    ->first(), true);
+                // Create new author row
+                $userGroup = new UserGroup([
+                    'roleId' => Role::ROLE_ID_AUTHOR,
+                    'contextId' => $contextId,
+                    'isDefault' => false,
+                    'showTitle' => false,
+                ]);
+                $userGroup->fill([
+                    'nameLocaleKey' => 'default.groups.name.author',
+                    'abbrevLocaleKey' => 'default.groups.abbrev.author',
+                    'abbrev' => Arr::mapWithKeys($supportedFormLocales, fn (string $locale) => [$locale => __('default.groups.abbrev.author', [], $locale)]),
+                    'name' => Arr::mapWithKeys($supportedFormLocales, fn (string $locale) => [$locale => __('default.groups.name.author', [], $locale)]),
+                ]);
+                // Save the new author instance to the database
+                $userGroup->save();
+            });
+
+
+        $translations = UserGroup::withRoleIds([Role::ROLE_ID_AUTHOR])
+            ->get()
+            ->map(fn (UserGroup $ug, int $ugId) => [
+                'userGroupId' => $ugId,
+                'key' => $ug->nameLocaleKey,
```

Comment:
Just a note, that nameLocaleKey does not need to exists, e.g. for the roles that were additionally created by journal manager